scripts/apps/vocabularies/directives/VocabularyConfigModalItems.jsx

```diff
@@ -48,7 +74,9 @@ export function VocabularyConfigModalItems(gettext) {
             // re-render on items changes
             scope.$watch('vocabulary.items', (items) => {
                 if (items && component) {
-                    component.setState({items});
+                    itemsValidation = validateItems(items);
+                    scope.itemsValidation.valid = validItems(itemsValidation);
```

Comment:
you can use [ngForm.$setValidity](https://docs.angularjs.org/api/ng/type/form.FormController#$setValidity) here so it works as other form controls